pipestat/backend.py

```diff
@@ -0,0 +1,1226 @@
+            record_identifier=record_identifier,
+            pipeline_type=pipeline_type,
+        ):
+            _LOGGER.error(f"Record '{record_identifier}' not found")
+            return False
+
+        if result_identifier and not self.check_result_exists(
+            result_identifier, record_identifier, pipeline_type=pipeline_type
+        ):
+            _LOGGER.error(f"'{result_identifier}' has not been reported for '{record_identifier}'")
+            return False
+
+        if rm_record:
+            self.remove_record(
+                record_identifier=record_identifier,
+                pipeline_type=pipeline_type,
+                rm_record=rm_record,
+            )
+        else:
+            val_backup = self._data[self.pipeline_name][pipeline_type][record_identifier][
+                result_identifier
+            ]
+            # self._data[self.project_name][pipeline_type][record_identifier][res_id] = val
+            del self._data[self.pipeline_name][pipeline_type][record_identifier][result_identifier]
+            _LOGGER.info(
+                f"Removed result '{result_identifier}' for record "
+                f"'{record_identifier}' from '{self.pipeline_name}' namespace"
+            )
+            if not self._data[self.pipeline_name][pipeline_type][record_identifier]:
+                _LOGGER.info(
+                    f"Last result removed for '{record_identifier}'. " f"Removing the record"
+                )
+                rm_record = True
+                self.remove_record(
+                    record_identifier=record_identifier,
+                    pipeline_type=pipeline_type,
+                    rm_record=rm_record,
+                )
+                # del self._data[self.project_name][pipeline_type][record_identifier]
+            with self._data as locked_data:
+                locked_data.write()
+        return True
+
+    def remove_record(
+        self,
+        record_identifier: Optional[str] = None,
+        pipeline_type: Optional[str] = None,
+        project_name: Optional[str] = None,
+        rm_record: Optional[bool] = False,
+    ) -> bool:
+        """
+        Remove a record, requires rm_record to be True
+
+        :param str record_identifier: unique identifier of the record
+        :param str result_identifier: name of the result to be removed or None
+             if the record should be removed.
+        :param str pipeline_type: "sample" or "project"
+        :param bool rm_record: bool for removing record.
+        :return bool: whether the result has been removed
+        """
+        if rm_record:
+            try:
+                _LOGGER.info(f"Removing '{record_identifier}' record")
+                del self._data[self.pipeline_name][pipeline_type][record_identifier]
+                with self._data as locked_data:
+                    locked_data.write()
+                return True
+            except:
+                # raise exception
```

Comment:
I do not understand this comment, can you clarify it or delete?

Reply:
Done. Added `LOGGER.warning`